CMake/Dependencies.cmake

```diff
@@ -23,7 +38,6 @@ else()
         # Aliasing local targets is not supported on CMake < 3.18, so make it global.
         set_target_properties(SDL2::SDL2-static PROPERTIES IMPORTED_GLOBAL TRUE)
       endif()
-      add_library(SDL2::SDL2 ALIAS SDL2::SDL2-static)

```

Comment:
Why remove this alias? Most of the CMake changes seem to come from removing this line.

Reply:
because both SDL_image and SDL_audiolib have such approach:
```
if(TARGET SDL2::SDL2)
    set(SDL_LIBRARIES SDL2::SDL2)
elseif(TARGET SDL2::SDL2-static)
    set(SDL_LIBRARIES SDL2::SDL2-static)
```
without this change - keeping the alias - those libraries will try to link with the dynamic SDL2 instead of the static one.

Reply:
If `SDL2::SDL2` is aliased to `SDL2::SDL2-static`, there is no dynamic SDL2 to begin with

Reply:
sorry, i saw the comment and i thought it was related to the the SDL2 `CMakeLists.txt` change
```
if(TARGET SDL2::SDL2-static)
  add_library(SDL2::SDL2 ALIAS SDL2-static)
endif()
```
i guess i have removed it during tests and forgot to add it back.

Reply:
ok, after reviewing the changes i remembered why it was removed. as it have removed the alias from the SDL2 CMakeLists i talked above the alias you are pointing is not needed anymore. and i can guarantee that there stills exits static linking as the xbox port is all static linked :)

Reply:
the issue was that we had both SDL2::SDL2 and SDL2::SDL2-static set up and it would be a problem for building SDL_image and SDL_audiolib as i mentioned before. so this way both static and dynamic linking should work properly.

Reply:
The way SDL2 is set up when using `add_subdirectory` is that only one of the `SDL2::SDL2` and `SDL2::SDL2-static` is available. When we add `SDL2` as an alias to `SDL2-static`, it does not mean that the library will be linked dynamically.

We do have other platforms, such as Vita and Switch, that are fully statically linked.

Reply:
Fine to keep it as is for now